js/ui/save-html.js

```diff
@@ -75,6 +80,17 @@ define(
                 }
                 ui.freshModal("Save Snapshot", $div);
             }
+        // Create and download an EPUB 3 version of the content
+        // Using (by default) the EPUB 3 conversion service set up at labs.w3.org/epub-generator
+        // For more details on that service, see https://github.com/iherman/respec2epub
+        ,   makeEPubHref: function(){
+                if (!conf.epubGenTool) {
+                    conf.epubGenTool = "https://labs.w3.org/epub-generator/cgi-bin/epub-generator.py";
+                }
+                var epubHref = conf.epubGenTool;
+                epubHref+= "?type=respec&url=" + doc.location.href;
```

Comment:
> On 08 Sep 2015, at 10:38 , Tobie Langel <notifications@github.com> wrote:
> 
> In js/ui/save-html.js <https://github.com/w3c/respec/pull/487#discussion_r38901622>:
> 
> > @@ -75,6 +80,17 @@ define(
> >                  }
> >                  ui.freshModal("Save Snapshot", $div);
> >              }
> > +        // Create and download an EPUB 3 version of the content
> > +        // Using (by default) the EPUB 3 conversion service set up at labs.w3.org/epub-generator
> > +        // For more details on that service, see https://github.com/iherman/respec2epub
> > +        ,   makeEPubHref: function(){
> > +                if (!conf.epubGenTool) {
> > +                    conf.epubGenTool = "https://labs.w3.org/epub-generator/cgi-bin/epub-generator.py";
> > +                }
> > +                var epubHref = conf.epubGenTool;
> > +                epubHref+= "?type=respec&url=" + doc.location.href;
> What I'm suggesting is simply:
> 
>  epubHref+= "?type=respec&url=" + encodeURIComponent(doc.location.href);

Not knowing the deep internals of respec: what exactly does it do? Just the percent encoding stuff? What does that buy me?


----
Ivan Herman, W3C
Digital Publishing Lead
Home: http://www.w3.org/People/Ivan/
mobile: +31-641044153
ORCID ID: http://orcid.org/0000-0003-0782-2704